tests/app/dao/test_ft_billing_dao.py

```diff
@@ -531,24 +552,30 @@ def test_fetch_billing_totals_for_year_variable_rates(notify_db_session):
     assert results[0].billable_units == 1
     assert results[0].chargeable_units == 1
     assert results[0].rate == Decimal('0.33')
+    assert results[0].free_allowance_remainder == 0
 
     assert results[1].notification_type == 'letter'
     assert results[1].notifications_sent == 1
     assert results[1].billable_units == 1
     assert results[1].chargeable_units == 1
     assert results[1].rate == Decimal('0.36')
+    assert results[1].free_allowance_remainder == 0
 
     assert results[2].notification_type == 'sms'
     assert results[2].notifications_sent == 1
     assert results[2].billable_units == 4
     assert results[2].chargeable_units == 4
     assert results[2].rate == Decimal('0.015')
+    # 4 units sent on the 16th, 0 on the 17th
+    assert results[2].free_allowance_remainder == 2
 
     assert results[3].notification_type == 'sms'
     assert results[3].notifications_sent == 2
     assert results[3].billable_units == 5
     assert results[3].chargeable_units == 5
     assert results[3].rate == Decimal('0.162')
+    # 1 free unit on the 16th, 1 on the 17th (+ 3 paid)
+    assert results[3].free_allowance_remainder == 0
```

Comment:
I'm a bit confused what the `free_allowance_remainder` is

the `notifications_sent`, `billable_units` and `chargeable_units` are all things that are counting to do with the `rate`. ie how many things you've sent at a certain rate. But it doesn't make intuitive sense to me to have a `free_allowance_remainder` attributed to a rate. I can see it only being attributed to a point in time, that is unless you are using the rate change as proxy for the date? If so, this seems like a particularly complex thing to get your head around?

This is just a bit of a quick comment before I rush off to meetings but hoping to get you thinking about it/replying in time for all my meetings to end if I'm lucky ;)

Reply:
(For anyone else reviewing, don't worry about this as it's gone.)

As discussed, I've removed this field / commit but added "free_chargeable_units" to the yearly usage API in the final commit, as we can still benefit from an explicitly named field to subtract with.